### DISCUSSION TITLE
Upgrade from 4.2.7 > 4.2.8 Compilation failed

### DISCUSSION THREAD

**Discussion by StrangeMinds1960** [mastodon/mastodon#29376] [Q&A] [answered]
mastodon@strangeminds:~/live$ RAILS_ENV=production bundle exec rails assets:precompile
Compiling...
Compilation failed:
`isModuleDeclaration` has been deprecated, please migrate to `isImportOrExportDeclaration`
    at isModuleDeclaration (/home/mastodon/live/node_modules/babel-plugin-lodash/node_modules/@babel/types/lib/validators/generated/index.js:2740:35)
    at PluginPass.Program (/home/mastodon/live/node_modules/babel-plugin-lodash/lib/index.js:102:44)
Browserslist: caniuse-lite is outdated. Please run:
  npx update-browserslist-db@latest
  Why you should do it regularly: https://github.com/browserslist/update-db#readme

Any clues as to where to find the problem?

**Comment by renchap** [comment]
Can you try to run `RAILS_ENV=production bundle exec rails console` and see if it displays any error?

The `isModuleDeclaration` and `Browserslist` messages are expected and not errors.

**Reply by StrangeMinds1960** [reply]
Waited a while, is this all I should expect?

mastodon@strangeminds:~/live$ RAILS_ENV=production bundle exec rails console
Chewy console strategy is `urgent`
Loading production environment (Rails 7.0.8.1)
irb(main):001:0>

**Comment by ClearlyClaire** [comment, marked as answer]
Those are just unrelated warnings. The error is webpack probably running out of memory. https://docs.joinmastodon.org/admin/troubleshooting/#i-encountered-a-compilation-error-while-executing-rails_envproduction-bundle-exec-rails-assetsprecompile-but-no-more-information-is-given-how-to-fix-it

**Reply by StrangeMinds1960** [reply]
Definitely a memory problem, I run Mastodon on quite a low spec VPS and will have to watch for this on future updates
